server/src/main/java/keywhiz/service/daos/GroupDAO.java

```diff
@@ -39,6 +43,10 @@
 import static keywhiz.jooq.tables.Secrets.SECRETS;
 
 public class GroupDAO {
+  private static final Long NO_OWNER = null;
+
+  private static final Groups GROUP_OWNERS = GROUPS.as("owners");
```

Comment:
Aliasing the owners table was the key insight into getting the self-join to work as intended.

Note that introducing an alias for GROUPS simplifies the generated SQL quite a bit (without an alias the fields are all prefixed by the schema and the table name), and we also have the option of invoking .asterisk() rather than .fields() to emit a literal asterisk into the query for all that table's fields. However, this only saves us a few bytes of query text and has some negative consequences as far as Jooq's type safety, I believe.